ingress-controller/internal/admission/handler.go

```diff
@@ -31,17 +32,105 @@ const (
 // RequestHandler is an HTTP server that can validate Kong Ingress Controllers'
 // Custom Resources using Kubernetes Admission Webhooks.
 type RequestHandler struct {
-	// Validator validates the entities that the k8s API-server asks
+	// validators validate the entities that the k8s API-server asks
 	// it the server to validate.
-	Validator KongValidator
-	// ReferenceIndexers gets the resources (KongPlugin and KongClusterPlugin)
-	// referring the validated resource (Secret) to check the changes on
-	// referred Secret will produce invalid configuration of the plugins.
-	ReferenceIndexers ctrlref.CacheIndexers
+	validators map[string]KongValidator
 
 	Logger logr.Logger
 }
 
+// dispatchValidationNoMatcher returns the first validator if exists, without any matching.
+// There is no specific matching further so it's safe to do it this way, see
+// - h.handleKongPlugin -> ValidatePlugin
+// - h.handleKongClusterPlugin -> ValidateClusterPlugin
+// - h.handleSecret -> ValidateCredential
+// - h.handleGateway -> ValidateGateway
+// - h.handleHTTPRoute -> ValidateHTTPRoute
```

Comment:
Can we please describe here what the return values mean? Specifically the `bool`